### PR TITLE
NBGL: Introduce nbgl_useCaseReviewStreamingXXX() API

### PR DESCRIPTION
**Pull request by xchapron-ledger**
## Description

NBGL: Introduce nbgl_useCaseReviewStreamingXXX() API

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: lib_nbgl/src/nbgl_use_case.c [hunk 1]
```diff
@@ -302,6 +313,81 @@ static const nbgl_content_t *getContentAtIdx(const nbgl_genericContents_t *gener
...
+                                   const char                 *reviewTitle,
+                                   const char                 *reviewSubTitle)
+{
+    centeredInfo->icon  = icon;
+    centeredInfo->text1 = reviewTitle;
+    centeredInfo->text2 = reviewSubTitle;
+#ifdef TARGET_STAX
+    centeredInfo->text3 = NULL;
+#else   // TARGET_STAX
+    centeredInfo->text3 = "Swipe to review";
```
Comment (cedelavergne-ledger) [inline lib_nbgl/src/nbgl_use_case.c:357]: Not sure, but you seem to consider the `#else` for Europa? Is this function also available for Nano devices?

**Comment by xchapron-ledger** [comment]
Example of usage can be found here: https://github.com/LedgerHQ/app-eos/pull/25/commits/d5919818ec7d963f1f0d1e862d22f8bef3ecef80

File: lib_nbgl/src/nbgl_use_case.c [hunk 2]
```diff
@@ -120,9 +121,18 @@ typedef struct {
...
+    const char                *finishTitle;
+    const nbgl_icon_details_t *icon;
+    uint8_t                    stepPageNb;
+} nbgl_reviewStreamingContext_t;
+
 typedef union {
     nbgl_homeAndSettingsContext_t homeAndSettings;
     nbgl_reviewContext_t          review;
+    nbgl_reviewStreamingContext_t reviewStreaming;
 } nbgl_BundleNavContext_t;
```
[the remark this answers is not in the document]
Comment (xchapron-ledger) [reply lib_nbgl/src/nbgl_use_case.c:136]: This is at another level, mainly above `GENERIC_NAV` which is one of the type of `NavType_t`.
And It is done with dedicated callbacks, so no need to store a type somewhere.

File: lib_nbgl/include/nbgl_use_case.h [hunk 3]
```diff
@@ -211,6 +211,20 @@ void nbgl_useCaseAddressReview(
 void nbgl_useCaseReviewStatus(nbgl_reviewStatusType_t reviewStatusType,
                               nbgl_callback_t         quitCallback);
 
+void nbgl_useCaseReviewStreamingStart(
+    nbgl_operationType_t       operationType,
+    const nbgl_icon_details_t *icon,
+    const char                *reviewTitle,
+    const char
+        *reviewSubTitle, /* Most often this is empty, but sometimes indicates a path / index */
```
[the remark this answers is not in the document]
Comment (xchapron-ledger) [reply lib_nbgl/include/nbgl_use_case.h]: Removed

**Review by nroggeman-ledger — APPROVED** [review]

**Review by apaillier-ledger — APPROVED** [review]

**Pull request merged**